File: app/src/main/java/com/iita/akilimo/models/RecommendationOptions.java
package com.iita.akilimo.models;

import androidx.annotation.NonNull;

import com.iita.akilimo.utils.enums.EnumAdviceTasks;


public class RecommendationOptions {

    private String recommendationName;
    private EnumAdviceTasks recCode;
    private int image;

    public RecommendationOptions() {
    }

    public RecommendationOptions(String recName, EnumAdviceTasks recommendationCode, int imageId) {
        this.setRecommendationName(recName);
        this.setRecCode(recommendationCode);
        this.setImage(imageId);
    }

    public String getRecommendationName() {
        return recommendationName;
    }

    public void setRecommendationName(@NonNull String recommendationName) {
        this.recommendationName = recommendationName;
    }

    public EnumAdviceTasks getRecCode() {
        return recCode;
    }

    public void setRecCode(EnumAdviceTasks recCode) {
        this.recCode = recCode;
    }

    public int getImage() {
        return image;
    }

    public void setImage(int image) {
        this.image = image;
    }
}

File: app/src/main/java/com/iita/akilimo/models/RecommendationOptions.kt
package com.iita.akilimo.models

import com.iita.akilimo.entities.AdviceStatus
import com.iita.akilimo.utils.enums.EnumAdviceTasks

class RecommendationOptions {
    var recName: String? = null
    var adviceName: EnumAdviceTasks? = null
    var statusImage: Int = 0
    var adviceStatus: AdviceStatus? = null

    constructor() {}

    @Deprecated("Removed this no need to pass image id now")
    constructor(recName: String, adviceName: EnumAdviceTasks, statusImage: Int) {
        this.recName = recName
        this.adviceName = adviceName
        this.statusImage = statusImage
    }

    constructor(
        recName: String,
        adviceName: EnumAdviceTasks,
        adviceStatus: AdviceStatus
    ) {
        this.recName = recName
        this.adviceName = adviceName
        this.statusImage = statusImage
        this.adviceStatus = adviceStatus
    }
}